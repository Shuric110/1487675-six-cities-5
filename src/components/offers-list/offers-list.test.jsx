import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter as Router} from 'react-router-dom';

import OffersList from "./offers-list";

const city = {
  id: 1,
  name: `Paris`,
  coordinates: {
    latitude: 48.8606146,
    longitude: 2.3354553
  },
  zoom: 12,
};

const offers = [
  {
    id: 1,
    city,
    pictures: [
      `/img/room.jpg`,
    ],
    placePicture: `/img/room.jpg`,
    coordinates: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198
    },
    mapZoom: 12,
    isPremium: false,
    isFavorite: true,
    nightlyCost: 80,
    title: `Wood and stone place`,
    type: `room`,
    rating: 4.2,
    description: `Excellent choice for travelers, offering many helpful amenities designed to enhance your stay.`,
    bedrooms: 1,
    maxAdults: 2,
    features: [
      `Wi-Fi`,
    ],
    host: {
      name: `Angelina`,
      avatar: `/img/avatar-angelina.jpg`,
      isSuper: true
    },
  },

  {
    id: 2,
    city,
    pictures: [
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/apartment-01.jpg`,
    ],
    placePicture: `/img/apartment-01.jpg`,
    coordinates: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198
    },
    mapZoom: 12,
    isPremium: true,
    isFavorite: false,
    nightlyCost: 132,
    title: `Canal View Prinsengracht`,
    type: `apartment`,
    rating: 3,
    description: `For travelers visiting Amsterdam, Canal View Prinsengracht is an excellent choice for rest and rejuvenation.
                  Well-known for its family-friendly environment and proximity to great restaurants and attractions, Canal View Prinsengracht makes it easy to enjoy the best of Amsterdam.`,
    bedrooms: 3,
    maxAdults: 5,
    features: [
      `Washing machine`,
      `Towels`,
      `Heating`,
    ],
    host: {
      name: `Max`,
      avatar: `/img/avatar-max.jpg`,
      isSuper: false
    },
  },
];

const noop = () => {};

jest.mock(`../bookmark-button/bookmark-button`, () => `BookmarkButton`);


describe(`Render OffersList`, () => {
  it(`OffersList renders correctly`, () => {
    const tree = renderer
      .create(
          <Router>
            <OffersList
              offers={offers}

              listClassName="list_class_1"
              itemClassName="item_class_1"
              setActiveOffer={noop}
              clearActiveOffer={noop}
            />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
