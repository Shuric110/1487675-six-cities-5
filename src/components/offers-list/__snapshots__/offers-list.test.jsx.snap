// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render OffersList OffersList renders correctly 1`] = `
<div
  className="list_class_1 places__list tabs__content"
>
  <article
    className="item_class_1 place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__image-wrapper"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            80
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <BookmarkButton
          baseClassName="place-card__bookmark-button"
          offer={
            Object {
              "bedrooms": 1,
              "city": Object {
                "coordinates": Object {
                  "latitude": 48.8606146,
                  "longitude": 2.3354553,
                },
                "id": 1,
                "name": "Paris",
                "zoom": 12,
              },
              "coordinates": Object {
                "latitude": 52.369553943508,
                "longitude": 4.85309666406198,
              },
              "description": "Excellent choice for travelers, offering many helpful amenities designed to enhance your stay.",
              "features": Array [
                "Wi-Fi",
              ],
              "host": Object {
                "avatar": "/img/avatar-angelina.jpg",
                "isSuper": true,
                "name": "Angelina",
              },
              "id": 1,
              "isFavorite": true,
              "isPremium": false,
              "mapZoom": 12,
              "maxAdults": 2,
              "nightlyCost": 80,
              "pictures": Array [
                "/img/room.jpg",
              ],
              "placePicture": "/img/room.jpg",
              "rating": 4.2,
              "title": "Wood and stone place",
              "type": "room",
            }
          }
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </BookmarkButton>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
        >
          Wood and stone place
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Private Room
      </p>
    </div>
  </article>
  <article
    className="item_class_1 place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="place-card__image-wrapper"
    >
      <a
        href="/offer/2"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            132
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <BookmarkButton
          baseClassName="place-card__bookmark-button"
          offer={
            Object {
              "bedrooms": 3,
              "city": Object {
                "coordinates": Object {
                  "latitude": 48.8606146,
                  "longitude": 2.3354553,
                },
                "id": 1,
                "name": "Paris",
                "zoom": 12,
              },
              "coordinates": Object {
                "latitude": 52.3909553943508,
                "longitude": 4.929309666406198,
              },
              "description": "For travelers visiting Amsterdam, Canal View Prinsengracht is an excellent choice for rest and rejuvenation.
                  Well-known for its family-friendly environment and proximity to great restaurants and attractions, Canal View Prinsengracht makes it easy to enjoy the best of Amsterdam.",
              "features": Array [
                "Washing machine",
                "Towels",
                "Heating",
              ],
              "host": Object {
                "avatar": "/img/avatar-max.jpg",
                "isSuper": false,
                "name": "Max",
              },
              "id": 2,
              "isFavorite": false,
              "isPremium": true,
              "mapZoom": 12,
              "maxAdults": 5,
              "nightlyCost": 132,
              "pictures": Array [
                "/img/apartment-03.jpg",
                "/img/studio-01.jpg",
                "/img/apartment-01.jpg",
              ],
              "placePicture": "/img/apartment-01.jpg",
              "rating": 3,
              "title": "Canal View Prinsengracht",
              "type": "apartment",
            }
          }
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </BookmarkButton>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
        >
          Canal View Prinsengracht
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
</div>
`;
